pymodbus/utilities.py

```diff
@@ -182,3 +186,11 @@ def hexlify_packets(packet):
     if not packet:
         return ""
     return " ".join([hex(int(x)) for x in packet])
+
+
+def system_endian():
```

Comment:
This looks wrong, because byte order can be different from word order.

It seems wrong that we add endian utilities, we should avoid that.

Reply:
So the issue here is that the array class which I'm using (because it provides a nice way to operate on 16 bit integer lists as bytes) encodes integers with the system native byte format. In pymodbus AFAIU our registers list is encoded as a list of big endian integers so when converting from a register list to a python array(which has a convenient bytes method) we have to conditionally byteswap when converting if our native system byteformat is little endian so that the individual python array elements get ordered as big endian bytes. So this system endianness check is only actually being used when encoding/decoding to the list of integers to/from bytes. We don't need to care about word order at this point either since we don't change word order when doing the register list to array to bytes conversions.

Once we are actually encoding/decoding integers i.e. in `_pack_words`/`_unpack_words` we can simply assume we are already big endian format and perform the byteswaping and wordswapping operations without having to care about system native byte format.